opentracing/src/main/java/opentracing/SpanPropagator.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package opentracing;
+
+import java.util.Map;
+
+/**
+ * Encodes or Decodes a {@linkplain Span} in binary or text formats.
+ *
+ * <p>The toXXX methods are expected to serialize spans into a pair of values representing
+ * separately the span / span identity, and the trace attributes. This is done specifically
+ * for binary protocols that may represent tracing identity in a dedicated fixed-length slot in the
+ * binary message format, so that it can be inspected efficiently by the middleware / routing layers
+ * without parsing the whole message.
+ */
+public interface SpanPropagator {
+
+  /**
+   * Encodes the span into a binary representation of the span's identity and trace
+   * attributes.
+   */
+  byte[] toBinary(Span span);
```

Comment:
Keep in mind that the only reason the separation exists is
 **if** the middleware exposes a binary stream and is using binary serialisation
 **if** the binary protocol uses fixed-length slots,
 **if** the IPC interaction wants to skip trace attributes
 **for** a small performance benefit or stability fix.

> This is done specifically
> for binary protocols that may represent tracing identity in a dedicated fixed-length slot in the
> binary message format, so that it can be inspected efficiently by the middleware / routing layers
> without parsing the whole message.


Surely, you can understand my questioning as to whether this really needs to belong in OpenTracing API v1.0

I'm trying to keep horizons as open as possible to cater for the adoption of as many tracing frameworks as possible. But it seems we're getting blind-sighted by niche performance optimisations for an additional DCP add-on… 

I'm in favour of clear documentation and Specification, and a simple API.

Using the extractor pattern it'ld be something along the lines of

         # Goal: framework encodes the Span into HTTP headers of the outbound request
        tracer.propagate(span, request)